packages/babel-helpers/src/helpers.js

```diff
@@ -428,10 +428,18 @@ helpers.objectDestructuringEmpty = template(`
 helpers.objectWithoutProperties = template(`
   (function (obj, keys) {
     var target = {};
-    for (var i in obj) {
-      if (keys.indexOf(i) >= 0) continue;
-      if (!Object.prototype.hasOwnProperty.call(obj, i)) continue;
-      target[i] = obj[i];
+    var objKeys = Object.keys(obj);
```

Comment:
Hm...

The first step in [DestructuringAssignmentEvaluation](https://tc39.github.io/proposal-object-rest-spread/#Rest-RuntimeSemantics-DestructuringAssignmentEvaluation) calls `RequireObjectCoercible`, which is also called by `Object.keys`... but [RestDestructuringAssignmentEvaluation](https://tc39.github.io/proposal-object-rest-spread/#Rest-RuntimeSemantics-RestDestructuringAssignmentEvaluation) calls [CopyDataProperties](https://tc39.github.io/proposal-object-rest-spread/#AbstractOperations-CopyDataProperties) which allows for `null`/`undefined` values (i.e. the things that `RequireObjectCoercible` would have rejected).

What do? :(

If we treat this specifically as an implementation of `CopyDataProperties`, it might be better to have `obj == null ? [] : Object.keys(obj)`.